dev/tests/integration/testsuite/Magento/CatalogInventory/Model/StockItemSave/StockItemDataChecker.php

```diff
@@ -74,6 +103,7 @@ public function checkStockItemData($sku, array $expectedData)
         $productLoadedByModel = $this->productFactory->create();
         $productLoadedByModel->load($product->getId());
         $this->doCheckStockItemData($product, $expectedData);
+        $this->checkIntegrityWithInventory($product, $expectedData);
```

Comment:
Need to remove this check
This is related only to MSI project

Reply:
@naydav 

this check fully relies on Legacy CatalogInventory
```
    /**
     * @param Product $product
     * @param array $expectedData
     * @return void
     */
    private function checkIntegrityWithInventory(Product $product, array $expectedData)
    {
        $searchCriteria = $this->searchCriteriaBuilder
            ->addFilter(SourceItemInterface::SOURCE_CODE, $this->defaultSourceProvider->getCode())
            ->addFilter(SourceItemInterface::SKU, $product->getSku())
            ->create();
        $sourceItems = $this->sourceItemRepository->getList($searchCriteria)->getItems();
        Assert::assertCount(1, $sourceItems);

        $sourceItem = reset($sourceItems);
        Assert::assertEquals(
            $expectedData[StockItemInterface::QTY],
            $sourceItem->getQuantity()
        );
        Assert::assertEquals(
            $expectedData[StockItemInterface::IS_IN_STOCK],
            (int)$sourceItem->getStatus()
        );
    }
```

there are no problems to have it in CE

Reply:
@maghamed $this->sourceItemRepository is instance of Magento\InventoryApi\Api\SourceItemRepositoryInterface.

Reply:
@naydav removed